[python-updates] Major updates 2025-08-05

mweinelt opened the pull request:
Jobset: https://hydra.lossy.network/jobset/nixpkgs/python-updates
Review-Reports: https://shells.darmstadt.ccc.de/~hexa/799.html

### Known issues:
- pytest 8.4 errors out on async functions when no async plugin is handling them
- pytest-asyncio 1.0 removes the `event_loop` fixture
- pytest-aiohttp needs to be overridden when pytest-asyncio_0 is required in parallel
- pytest 8.3 was introduced to counter ` Failed: 'yield' keyword is allowed in fixtures, but not in tests`

DavHau committed:
python3Minimal: refactor - add withMinimal flag

simplifies the interface of building cpython minimally

DavHau committed:
python3Minimal: remove pkg-config dependency

Removes the build time dependency on pkg-config if python3 is build with the `withMinimal` flag enabled

This change is driven by the motivation to use python3 earlier in stdenv for hooks.

mweinelt committed:
python3Packages.thriftpy2: 0.5.2 -> 0.5.3

https://github.com/Thriftpy/thriftpy2/blob/v0.5.3/CHANGES.rst

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.tifffile: 2025.5.10 -> 2025.6.11

https://github.com/cgohlke/tifffile/blob/v2025.6.11/CHANGES.rst

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.tika-client: 0.9.0 -> 0.10.0

https://github.com/stumpylog/tika-client/blob/0.10.0/CHANGELOG.md

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.tianshou: 1.1.0 -> 1.2.0

https://github.com/thu-ml/tianshou/releases/tag/v1.2.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.tiler: 0.5.7 -> 0.6.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.tiledb: 0.33.2 -> 0.34.2

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.tld: 0.13 -> 0.13.1

https://github.com/barseghyanartur/tld/blob/0.13.1/CHANGELOG.rst

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.total-connect-client: 2025.1.4 -> 2025.5

https://github.com/craigjmidwinter/total-connect-client/releases/tag/2025.5

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.torchio: 0.20.17 -> 0.20.19

https://github.com/TorchIO-project/torchio/blob/v0.20.19/CHANGELOG.md

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.tox: 4.26.0 -> 4.28.4

https://github.com/tox-dev/tox/releases/tag/4.28.4

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.trackpy: 0.6.4 -> 0.7

https://github.com/soft-matter/trackpy/releases/tag/v0.7

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.trakit: 0.2.2 -> 0.2.5

https://github.com/ratoaq2/trakit/releases/tag/0.2.5

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.tree-sitter-markdown: 0.3.2 -> 0.5.0

https://github.com/tree-sitter-grammars/tree-sitter-markdown/releases/tag/v0.5.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.tree-sitter-rust: 0.23.2 -> 0.24.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.tree-sitter: 0.24.0 -> 0.25.0

https://github.com/tree-sitter/py-tree-sitter/releases/tag/v0.25.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.treq: 24.9.1 -> 25.5.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.trino-python-client: 0.323.0 -> 0.334.0

https://github.com/trinodb/trino-python-client/blob/0.334.0/CHANGES.md

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.treelib: 1.7.1 -> 1.8.0

https://github.com/caesar0301/treelib/releases/tag/v1.8.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.trl: 0.19.0 -> 0.20.0

https://github.com/huggingface/trl/releases/tag/v0.20.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.trytond: 7.6.3 -> 7.6.5

https://foss.heptapod.net/tryton/tryton/-/blob/trytond-7.6.5/trytond/CHANGELOG?ref_type=tags

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.twisted: 24.11.0 -> 25.5.0

https://github.com/twisted/twisted/blob/twisted-25.5.0/NEWS.rst

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.ttkbootstrap: 1.12.1 -> 1.14.2

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.type-infer: 0.0.21 -> 0.0.23

https://github.com/mindsdb/type_infer/releases/tag/v0.0.23

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.typed-settings: 24.6.0 -> 25.0.0

https://gitlab.com/sscherfke/typed-settings/-/blob/25.0.0/CHANGELOG.rst

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.typeguard: 4.4.2 -> 4.4.4

https://github.com/agronholm/typeguard/releases/tag/4.4.4

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.types-colorama: 0.4.15.20240311 -> 0.4.15.20250801

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.types-lxml: 2025.03.04 -> 2025.03.30

https://github.com/abelcheung/types-lxml/releases/tag/2025.03.30

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.types-protobuf: 5.29.1.20250315 -> 6.30.2.20250703

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.types-python-dateutil: 2.9.0.20241206 -> 2.9.0.20250708

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.types-psutil: 7.0.0.20250401 -> 7.0.0.20250801

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.types-setuptools: 78.1.0.20250329 -> 80.9.0.20250801

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.types-pyyaml: 6.0.12.20250402 -> 6.0.12.20250516

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.typeshed-client: 2.7.0 -> 2.8.2

https://github.com/JelleZijlstra/typeshed_client/releases/tag/v2.8.2

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.tyro: 0.9.19 -> 0.9.27

https://github.com/brentyi/tyro/releases/tag/v0.9.27

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.ueberzug: 18.1.9 -> 18.3.1

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.uart-devices: 0.1.0 -> 0.1.1

https://github.com/bdraco/uart-devices/blob/v0.1.1/CHANGELOG.md

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.uharfbuzz: 0.45.0 -> 0.51.1

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.ufo2ft: 3.5.1 -> 3.6.0

https://github.com/googlefonts/ufo2ft/releases/tag/v3.6.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.ultralytics: 8.3.143 -> 8.3.174

https://github.com/ultralytics/ultralytics/releases/tag/v8.3.174

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.umap-learn: 0.5.8 -> 0.5.9.post2

https://github.com/lmcinnes/umap/releases/tag/release-release-0.5.9.post2

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.unidata-blocks: 0.0.16 -> 0.0.17

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.unsloth: 2025.6.5 -> 2025.8.1

https://github.com/unslothai/unsloth/releases/tag/2025.8.1

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.unsloth-zoo: 2025.6.4 -> 2025.8.1

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.unstructured-client: 0.38.1 -> 0.42.0

https://github.com/Unstructured-IO/unstructured-python-client/blob/v0.42.0/RELEASES.md

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.upcloud-api: 2.6.0 -> 2.7.0

https://github.com/UpCloudLtd/upcloud-python-api/blob/v2.7.0/CHANGELOG.md

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.unstructured: 0.17.2 -> 0.18.12

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.uqbar: 0.7.4 -> 0.9.5

https://github.com/josiah-wolf-oberholtzer/uqbar/releases/tag/v0.9.5

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.uritemplate: 4.1.1 -> 4.2.0

https://github.com/python-hyper/uritemplate/blob/4.2.0/HISTORY.rst

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.urllib3: 2.4.0 -> 2.5.0

https://github.com/urllib3/urllib3/blob/2.5.0/CHANGES.rst

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.urwid: 2.6.16 -> 3.0.2

https://github.com/urwid/urwid/releases/tag/3.0.2

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.uuid6: 2024.7.10 -> 2025.0.1

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.urwidtrees: 1.0.3 -> 1.0.4

https://github.com/pazz/urwidtrees/releases/tag/1.0.4

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.uxsim: 1.7.2 -> 1.8.2

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.uvicorn: 0.34.2 -> 0.35.0

https://github.com/encode/uvicorn/blob/0.35.0/CHANGELOG.md

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.vfblib: 0.9.5 -> 0.10.2

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.versioningit: 3.1.2 -> 3.3.0

https://versioningit.readthedocs.io/en/latest/changelog.html

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.viewstate: 0.6.0 -> 0.7.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.virtualenv: 20.31.2 -> 20.33.1

https://github.com/pypa/virtualenv/blob/20.33.1/docs/changelog.rst

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.viser: 1.0.0 -> 1.0.3

https://github.com/nerfstudio-project/viser/releases/tag/v1.0.3

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.wagtail: 6.4.1 -> 7.1

https://github.com/wagtail/wagtail/blob/v7.1/CHANGELOG.txt

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.weasyprint: 65.1 -> 66.0

https://github.com/Kozea/WeasyPrint/releases/tag/v66.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.weaviate-client: 4.12.0 -> 4.16.5

https://github.com/weaviate/weaviate-python-client/releases/tag/v4.16.5

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.weatherflow4py: 1.3.1 -> 1.4.1

https://github.com/jeeftor/weatherflow4py/releases/tag/v1.4.1

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.webssh: 1.6.2 -> 1.6.3

https://github.com/huashengdun/webssh/releases/tag/v1.6.3

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.webtest: 3.0.4 -> 3.0.6

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.wgpu-py: 0.22.2 -> 0.23.0

https://github.com/pygfx/wgpu-py/blob/v0.23.0/CHANGELOG.md

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.whisperx: 3.3.2 -> 3.4.2

https://github.com/m-bain/whisperX/releases/tag/v3.4.2

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.wn: 0.11.0 -> 0.13.0

https://github.com/goodmami/wn/blob/v0.13.0/CHANGELOG.md

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.wrf-python: 1.3.4.1 -> 1.4.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.xarray: 2025.04.0 -> 2025.07.1

https://github.com/pydata/xarray/blob/v2025.07.1/doc/whats-new.rst

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.x-transformers: 2.5.6 -> 2.6.1

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.xgrammar: 0.1.19 -> 0.1.22

https://github.com/mlc-ai/xgrammar/releases/tag/v0.1.22

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.xmlschema: 4.0.1 -> 4.1.0

https://github.com/sissaschool/xmlschema/blob/v4.1.0/CHANGELOG.rst

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.xmlsec: 1.3.15 -> 1.3.16

https://github.com/xmlsec/python-xmlsec/releases/tag/1.3.16

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.xvfbwrapper: 0.2.10 -> 0.2.13

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.xyzservices: 2025.1.0 -> 2025.4.0

https://github.com/geopandas/xyzservices/releases/tag/2025.4.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.youseedee: 0.6.0 -> 0.7.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.youtube-transcript-api: 1.0.3 -> 1.2.2

https://github.com/jdepoix/youtube-transcript-api/releases/tag/v1.2.2

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.z3c-checkversions: 2.1 -> 3.0

https://github.com/zopefoundation/z3c.checkversions/blob/3.0/CHANGES.rst

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.zdaemon: 5.1 -> 5.2.1

https://github.com/zopefoundation/zdaemon/blob/5.2.1/CHANGES.rst

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.zimports: 0.6.1 -> 0.6.2

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.zipp: 3.21.0 -> 3.23.0

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.zstd: 1.5.6.6 -> 1.5.7.2

This commit was automatically generated using update-python-libraries.

mweinelt committed:
python3Packages.setuptools-scm: 8.3.1 -> 9.0.1

https://github.com/pypa/setuptools_scm/blob/9.0.1/CHANGELOG.md

mweinelt committed:
python3Packages.urllib3: unpin setuptools-scm

sternenseemann commented:
Let me know about any regressions related to fonttools and friends. I checked the reverse dependencies, but may have missed some of them since so much depends on this stuff.

panicgh committed:
python3Packages.pynitrokey: 0.9.3 -> 0.10.0

https://github.com/Nitrokey/pynitrokey/releases/tag/v0.10.0

mweinelt committed:
python3Packages.cachetools: 5.5.2 -> 6.1.0

https://github.com/tkem/cachetools/blob/v6.1.0/CHANGELOG.rst

mweinelt committed:
python3Packages.nbformat: ignore unraisable exceptions during tests

mweinelt committed:
python3Packages.fnv-hash-fast: relax fnvhash constraint

mweinelt committed:
python3Packages.pytest-aiohttp: ignore pytest deprecation warnings

mweinelt committed:
python3Packages.psycopg: relax deps and clean up

panicgh committed:
python3Packages.fido2: skip device tests (#431308)

By default, unit tests include hardware authenticator device tests. They
hang nixpkgs builds for a long time waiting for a touch of the
authenticator device. Skip such tests to avoid waiting.

mweinelt committed:
python3Packages.oauthlib: disable failing test

mweinelt committed:
python3Packages.pytest-benchmark: remove xdist

This causes internal errors with pytest 8.4.

mweinelt committed:
python3Packages.black: fix click 8.2 compat

mweinelt committed:
python3Packages.requests-oauthlib: disable failing test

mweinelt committed:
python3Packages.volvooncall: pin to pytest-asyncio 0.x

mweinelt committed:
python3Packages.google-auth: relax cachetools constraint

mweinelt committed:
python3Packages.aws-xray-sdk: pin pytest-asyncio 0.x and disable failing tests

mweinelt committed:
Revert "python3Packages.click: 8.1.8 -> 8.2.2"

This reverts commit ab14a5defe0b15296f6abd1dbf4f8328472ceaea.

Backed out due to breakages in click-repl, which breaks celery.

mweinelt committed:
python3Packages.remarshal: add testing hint

mweinelt committed:
python3Packages.audioop-lts: 0.2.1 -> 0.2.2

https://github.com/AbstractUmbra/audioop/releases/tag/0.2.2

mweinelt committed:
python3Packages.aresponses: pin pytest-asyncio 0.x

mweinelt committed:
python3Packages.backoff: pin to pytest-asyncio 0.x

mweinelt committed:
python3Packages.pytest-bdd: pin to pytest7

This package requires an update, which depends on the introduction of the
gherkin parser/compiler (https://github.com/cucumber/gherkin).

mweinelt committed:
python3Packages.remarshal: 1.0.0 -> 1.0.1

https://github.com/remarshal-project/remarshal/releases/tag/v1.0.1

mweinelt committed:
python3Packages.promise: disable failing test

mweinelt committed:
python3Packages.botocore: 1.38.32 -> 1.40.4

https://github.com/boto/botocore/blob/1.40.4/CHANGELOG.rst

mweinelt committed:
python3Packages.s3transfer: 0.13.0 -> 0.13.1

https://github.com/boto/s3transfer/blob/0.13.1/CHANGELOG.rst

mweinelt committed:
awscli: 1.40.31 -> 1.42.4

https://github.com/aws/aws-cli/blob/1.42.4/CHANGELOG.rst

mweinelt committed:
python3Packages.terminado: ignore unraisable exceptions during tests

mweinelt committed:
python3Packages.pytest-describe: modernize, pin to pytest7

mweinelt committed:
python3Packages.dj-rest-auth: disable failing tests

mweinelt committed:
pretalx: pin django-csp at 3.8

mweinelt committed:
borgbackup: relax msgpack constraint

mweinelt committed:
python3Packages.pyopenxl: update disabled tests

mweinelt committed:
Revert "python3Packages.pip-tools: 7.4.1 -> 7.5.0"

This reverts commit 8ae23cfe65db122a3360ee10995d5caa38b4b421.

Breaks home-assistant-chip-wheels.

mweinelt committed:
python3Packages.hijridate: rename from hijri-converter;  2.3.2.post1 -> 2.5.0

mweinelt committed:
esphome: relax wheel pin

mweinelt committed:
python3Packages.rapidfuzz: relax cython constraint

mweinelt committed:
python3Packages.peewee: 3.18.1 -> 3.18.2

https://github.com/coleifer/peewee/blob/3.18.2/CHANGELOG.md

mweinelt committed:
python3Packages.virtualenvwrapper: fix dist-info version

mweinelt committed:
python3Packages.nettigo-air-monitor: fix dist-info version

mweinelt committed:
python3Packages.aiowebostv: fix dist-info version

mweinelt committed:
python3Packages.autobahn: pin to pytest-asyncio 0.x

mweinelt committed:
python3Packages.yangson: relax elementpath constraint

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/788.html

mweinelt committed:
python3Packages.aiobotocore: disable failing test

mweinelt committed:
python3Packages.django_5_2: 5.2.4 -> 5.2.5

https://docs.djangoproject.com/en/5.2/releases/5.2.5/

mweinelt committed:
python3Packages.sanic: disable failing tests

mweinelt committed:
python3Packages.python-socketio: pin to older pytest

The test are very async heavy, but don't declare async markers, which
breaks with pytest 8.4.

sternenseemann committed:
python3Packages.openstep-plist: add patch for CPython >= 3.1 compat

mweinelt committed:
python3Packages.ansible-core: fix build

mweinelt committed:
python3Packages.zarr: 2.18.7 -> 3.1.1

https://zarr.readthedocs.io/en/stable/release-notes.html

mweinelt committed:
python3Packages.icalendar: disable failing tests

mweinelt committed:
python3Packages.pyee: pin pytest-asyncio 0.x

mweinelt committed:
python3Packages.omegaconf: downgrade pytest, ignore userwarning

mweinelt committed:
python3Packages.statsmodels: relax setuptools-scm constraint

mweinelt committed:
python3Packages.weconnect: relax oauthlib constraint

mweinelt committed:
python3Packages.markdown: downgrade pytest

test/test_markdown2.py - Failed: 'yield' keyword is allowed in fixtures, but not in tests (test_cases)

mweinelt committed:
python3Packages.levenshtein: relax cython constraint

mweinelt committed:
python3Packages.dependency-groups: init at 1.3.1

mweinelt committed:
python3Packages.geopy: downgrade pytest

Uses undeclared async test functions which break with pytest 8.4.

mweinelt committed:
python3Packages.webtest-aiohttp: fix tests with pytest8.4

mweinelt committed:
python3Packages.cherrypy: ignore unraisable exceptions during tests

mweinelt committed:
python3Packages.makefun: downgrade pytest

Uses undeclared async functions, which breaks with pytest 8.4.

mweinelt committed:
python3Packages.xmldiff: disable failing test

mweinelt committed:
python3Packages.gql: pin to pytest-asyncio 0.x

mweinelt committed:
python3Packages.anova-wifi: disable failing test

Fails with pytest 8.4.

mweinelt committed:
python3Packages.pyatv: pin pytest-asyncio at 0.x

Otherwise it won't find the `event_loop` fixture.

mweinelt committed:
python3Packages.pyaprilaire: pin pytest-asyncio at 0.x

mweinelt committed:
python3Package.pytest_8_3: init at 8.3.5

mweinelt committed:
python3Packages.apischema: pin to pytest 8.3.x

mweinelt committed:
python3Packages.inkex: relax lxml constraint

mweinelt committed:
python3Packages.pytest-examples: disable failing tests

kirillrdy committed:
python3Packages.fabric: disable failing tests

mweinelt committed:
python3Packages.aioboto3: make compatible with boto3 1.40.4

sternenseemann commented:
ttfautohint-py is broken because of a switch of (bundled?) library to a bundled executable: https://github.com/fonttools/ttfautohint-py/commit/1a84788cfe512c612ced2dcdf83be6750fbc6f05

mweinelt committed:
python3Packages.rencode: unstable-2021-08-10 -> 1.0.8

https://github.com/aresch/rencode/releases/tag/v1.0.7
https://github.com/aresch/rencode/releases/tag/v1.0.8

mweinelt committed:
python3Packages.apscheduler: pin pytest 8.3

mweinelt committed:
python3Packages.sshtunnel: mark broken

mweinelt committed:
python3Packages.matrix-nio: pin pytest-asyncio to 0.x

mweinelt committed:
python3Packages.caio: pin pytest 8.3, pytest-asycio 0.x

mweinelt committed:
python3Packages.grpclib: pin pytest-asyncio 0.x

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/799.html

mweinelt commented:
> * [ ]  ttfautohint-py is broken because of a switch of (bundled?) library to a bundled executable: [fonttools/ttfautohint-py@1a84788](https://github.com/fonttools/ttfautohint-py/commit/1a84788cfe512c612ced2dcdf83be6750fbc6f05)

Backed out of this update.

mweinelt committed:
python3Packages.zigpy: pin tests to pytest-asyncio 0.x

mweinelt committed:
pdm: 2.24.2 -> 2.25.5

https://github.com/pdm-project/pdm/releases/tag/2.25.0
https://github.com/pdm-project/pdm/releases/tag/2.25.1
https://github.com/pdm-project/pdm/releases/tag/2.25.2
https://github.com/pdm-project/pdm/releases/tag/2.25.3
https://github.com/pdm-project/pdm/releases/tag/2.25.4
https://github.com/pdm-project/pdm/releases/tag/2.25.5

mweinelt committed:
python3Packages.django*: fix compat with python 3.13.6 and newer

mweinelt committed:
python313Packages.websockets: temporarily disable tests

mweinelt committed:
thrift: wrap python environment and provide six

mweinelt committed:
python3Packages.django-allauth: 65.9.0 -> 65.10.0

https://codeberg.org/allauth/django-allauth/src/tag/65.10.0/ChangeLog.rst

mweinelt committed:
python3Packages.django-allauth: backport patch for dj-rest-auth compat

mweinelt committed:
libplacebo: fix codegen with python 3.13.6

mweinelt committed:
python3Packages.fido2_2: drop

The fido2 package is now on 2.0.

mweinelt committed:
paperless-ngx: pin fido2 at 1.2.0

mweinelt committed:
qemu: wrap python with distlib

```
mkvenv: Creating non-isolated virtual environment at 'pyvenv'

*** Ouch! ***

found no usable distlib, please install it
```

mweinelt committed:
python3Packages.curio: disable stuck threading test

mweinelt committed:
python3Packages.django-tasks: disable failing test

mweinelt committed:
python3Packages.python-box: disable failing test

mweinelt committed:
python3Packages.s3fs: test with pytest-asyncio

mweinelt committed:
python3Packages.tesserocr: relax cython constraint

mweinelt committed:
python3Packages.django-debug-toolbar: 5.0.1 -> 6.0.0

https://django-debug-toolbar.readthedocs.io/en/latest/changes.html

mweinelt committed:
python3Packages.django-graphiql-debug-toolbar: disable tests

Not sure if the package still works. Upstream looks dead.

mweinelt committed:
weblate: relax lxml constraint

mweinelt committed:
python3Packages.anywidget: disable failing tests

mweinelt committed:
python313Packages.pychromecast: relax wheel constraint

mweinelt committed:
python3Packages.alive-progress: relax about-time constraint

mweinelt committed:
python3Packages.manimpango: relax cython constraint

mweinelt committed:
python3Packages.karton-core: relax aioboto3 constraint

mweinelt committed:
python3Packages.pyvisa: 1.14.1 -> 1.15.0

https://github.com/pyvisa/pyvisa/compare/refs/tags/1.14.1...refs/tags/1.15.0

mweinelt committed:
python3Packages.ha-silabs-firmware-client: test with pytest-asyncio

mweinelt committed:
python3Packages.adlfs: relax azure-datalake-store constraint

mweinelt committed:
python3Packages.aiorpcx: pin pytest-asyncio 0.x

mweinelt committed:
python3Packages.pytest-factoryboy: 2.6.1 -> 2.8.1

https://github.com/pytest-dev/pytest-factoryboy/compare/2.6.1...2.8.1

mweinelt committed:
python3Packags.hydra-core: pin pytest 8.3

mweinelt committed:
python3Packages.llm-echo: test with pytest-asyncio

mweinelt committed:
python3Packages.flask-restful: pin pytest 8.3

mweinelt committed:
devpi-server: test with pytest-asyncio

mweinelt committed:
python3Packages.databricks-sql-connector: relax pandas constraint

mweinelt committed:
python3Packages.google-cloud-artifact-registry: test with pytest-asyncio

mweinelt committed:
python3Packages.rasterio: relax cython constraint

mweinelt committed:
python3Packages.django-q2: unpin hiredis, use redisTestHook

mweinelt committed:
mailman-web: drop patches on django-allauth, unpin hiredis

mweinelt committed:
python3Packages.scooby: disable failing test

mweinelt committed:
python3Packages.esprima: pin to pytest 8.3

mweinelt committed:
python3Packages.colormath: pin to pytest 8.3

mweinelt committed:
python3Packages.comicon: relax ebooklib constraint

mweinelt committed:
python3Packages.cohere: relax httpx-sse constraint

mweinelt committed:
python3Packages.rapidgzip: relax cython constraint

mweinelt committed:
python3Packages.htmldate: relax lxml constraint

mweinelt committed:
python3Packages.preggy: pin to pytest 8.3, modernize

mweinelt committed:
python3Packages.zha: test with pytest-asyncio 0.x

mweinelt committed:
python3Packages.paste: ignore deprecation warning in tests

mweinelt committed:
python3Packages.blis: disable failing tests

mweinelt committed:
badsecret: relax viewstate constraint

mweinelt committed:
python3Packages.dbt-protos: init at 1.0.351

mweinelt committed:
python3Packages.fiona: relax cython constraint

mweinelt committed:
python3Packages.dbt-adapters: 1.14.8 -> 1.16.3

https://github.com/dbt-labs/dbt-adapters/blob/main/dbt-adapters/CHANGELOG.md

mweinelt committed:
python3Packages.sqlglot: 26.16.2 -> 27.6.0

https://github.com/tobymao/sqlglot/blob/v27.6.0/CHANGELOG.md

mweinelt committed:
python3Packages.aioasuswrt: test with pytest-asyncio 0.x

mweinelt committed:
python3Packages.papermill: disable failing test

This is due to API compat issues with azure data lake.

mweinelt committed:
python3Packages.sqlframe: 3.38.0 -> 3.38.2

https://github.com/eakmanrq/sqlframe/releases/tag/v3.38.1
https://github.com/eakmanrq/sqlframe/releases/tag/v3.38.2

mweinelt committed:
python3Packages.jsonrpc-websocekt: test with pytest-asyncio 0.x

mweinelt committed:
python3Packages.aiolookin: disable tests

They're async, have previously been skipped and are now erroring out.

mweinelt committed:
python3Packages.aiohttp-sse: test with pytest-asyncio 0.x

mweinelt committed:
python3Packages.async-modbus: test with pytest-asyncio

mweinelt committed:
python3Packages.aionotion: relax various dependencies

mweinelt committed:
python3Packages.opower: fix tests

mweinelt committed:
python3Packages.pyrainbird: test with pytest-asyncio 0.x

mweinelt committed:
python3Packages.pysiaalarm: fix tests

mweinelt committed:
python3Packages.pytest-mockservers: disable failing test

The event_loop fixutre was deprecated in pytest-asyncio 0.22 and removed
in 1.0. Since pytest-mockservers propagates pytest-asyncio we must not
propagate anything but the default pytest-asyncio package.

mweinelt committed:
python3Packages.thermopro-ble: fix tests

mweinelt committed:
python3Packages.uasiren: fix tests

mweinelt committed:
python3Packages.xbox-webapi: fix tests, modernize

mweinelt committed:
python3Packages.azure-kusto-ingest: relax deps

mweinelt committed:
python3Packages.python-awair: disable tests

Can't get these to run, likely requires a pytest-aiohttp update.

mweinelt committed:
python3Packages.pybalboa: fix tests

mweinelt committed:
python3Packages.pydaikin: disable failing async tests

mweinelt committed:
python3Packages.aioeafm: disable tests

All are async and not properly configured for pytest-asyncio.

mweinelt committed:
python3Packages.aioecowitt: fix tests with pytest-asyncio 0.x

mweinelt committed:
python3Packages.afsapi: disable tests

mweinelt committed:
python3Packages.fyta-cli

mweinelt committed:
python3Packages.geocachingapi: relax reverse-geocode constraint

mweinelt committed:
python3Packages.aiohomekit: test with pytest-asyncio 0.x

mweinelt committed:
python3Packages.hyperion-py: test with pytest-asyncio 0.x

mweinelt committed:
python3Packages.devolo-plc-api: test with pytest-asyncio 0.x

mweinelt committed:
python3Packages.aioimaplib: test with pytest-asyncio 0.x

mweinelt committed:
python3Packages.inkbird-ble: run async tests

mweinelt committed:
python3Packages.pykaleidescape: test with pytest-asyncio 0.x

mweinelt committed:
python3Packages.xknx: test with pytest-asyncio 0.x

mweinelt committed:
python3Packages.langchain-tests: pin pytest-asyncio 0.x
